### ISSUE TITLE
Support for boolean values in map transform in compositions

### ISSUE THREAD

**Issue by samidbb** [crossplane/crossplane#2588]
### What problem are you facing?
We are building a package that offers claim to create sqldatabase instances on AWS on-demand (dynamic provisioning).
User has to specify boolean parameter `isProduction` to tell whether the instance is running in production  mode or in non-production mode. Based on the  provided value, `skipFinalSnapshotBeforeDeletion` will be set.

XRD:
```yaml
...
    schema:
      openAPIV3Schema:
        type: object
        properties:
          spec:
            type: object
            properties:
              parameters:
                type: object
                properties:
                  isProduction:
                    description: Specify whether it is intended to run in production environment or not
                    type: boolean
...
```

The logic should be as follows:
if isProduction is true then we want to pass false to skipFinalSnapshotBeforeDeletion
if isProduction is false then we want to pass true to skipFinalSnapshotBeforeDeletion

The challenge is that crossplane composition logic does not have a map that supports boolean types and there is no transform for inverting boolean either.

So I ended up having to convert boolean to string to be able to make use of the map transform, then convert back to boolean to be able to pass the value that I want pass to the provider:

Composition:
```yaml
...
    - fromFieldPath: "spec.parameters.isProduction"
      toFieldPath: "spec.forProvider.skipFinalSnapshotBeforeDeletion"
      transforms:
      - type: convert
        convert:
          toType: string
      - type: map
        map:
          "true": "false"
          "false": "true"
      - type: convert
        convert:
          toType: bool
...
```

Which does the job but I think it's a bit too much code to handle a simple case.

### How could Crossplane help solve your problem?

[I asked the community about this on Slack](https://crossplane.slack.com/archives/CEG3T90A1/p1632150509490500), and @negz suggested to raise an issue to investigate whether making the map type configurable or just include a specific transform for inverting bool for solving this issue.

A map transform that supports boolean types will definitely help solve this case and many similar cases that we are going to have in our journey for building configuration packages in our organization. 

The above code could have easily been reduced to this:
```yaml
...
    - fromFieldPath: "spec.parameters.isProduction"
      toFieldPath: "spec.forProvider.skipFinalSnapshotBeforeDeletion"
      transforms:
      - type: map
        map:
          true: false
          false: true
...
```

An inverting tool could make the logic more clean. So maybe this could be written in this way instead:
```yaml
...
    - fromFieldPath: "spec.parameters.isProduction"
      toFieldPath: "spec.forProvider.skipFinalSnapshotBeforeDeletion"
      transforms:
      - type: bool
        bool:
          invert: %s
...
```

**Comment by mcbenjemaa** [comment]
@negz Can you share your thoughts regarding this request?
and if applicable can you describe how the API should look like, from a Design perspective ?

If so, and I have a little time, I'd work on this issue.

**Comment by urbaniak** [comment]
I'm having similar issue - trying to set bool flags on s3 bucket based on `spec.access` field which is a string private/public using that patch and it's not populating the `restrictPublicBuckets` field

```yaml
        - type: FromCompositeFieldPath
          fromFieldPath: spec.access
          toFieldPath: spec.forProvider.restrictPublicBuckets
          transforms:
            - type: map
              map:
                private: "true"
                public: "false"
            - type: convert
              convert:
                toType: bool
```

that one (without a transformation, in the same patch) is working fine

```yaml
        - type: FromCompositeFieldPath
          fromFieldPath: spec.access
          toFieldPath: spec.forProvider.acl
          transforms:
            - type: map
              map:
                private: "private"
                public: "public-read"
```

**Comment by MisterMX** [comment]
@hasheddan I believe this has been resolved with #3150. Or are we still missing something?

**Comment by hasheddan** [comment]
Ah nice, looks like we are all good here. Thanks @MisterMX !
